build.gradle

```diff
@@ -63,13 +63,19 @@ allprojects {
     }
 
     tasks.withType(Test) {
+        jvmArgumentProviders.add(project.ext.cliArgProvider({
+            [
+                    "-Dcom.linecorp.armeria.blockhound.reportFile=${project.buildDir}/blockhound.log",
+                    "-XX:HeapDumpPath=${project.buildDir}",
```

Comment:
My suggestion would be to not overgeneralize the `cliArgProvider`, but instead create one specific for tests, set the `blockhound` as an output file with relative path sensitivity and the `HeapDumpPath` as `@LocalState`, which will also get removed if the task is `FROM-CACHE`.